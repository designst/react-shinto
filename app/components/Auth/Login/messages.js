/**
 * Login Messages
 *
 * This contains all the text for the Login component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'app.components.Login.header',
    defaultMessage: 'This is the Login component!',
  },
  action: {
    id: 'app.components.Login.action',
    defaultMessage: 'Login',
  },
  username: {
    label: {
      id: 'app.components.Login.username.label',
      defaultMessage: 'Username',
    },
    placeholder: {
      id: 'app.components.Login.username.placeholder',
      defaultMessage: 'Username',
    },
    help: {
      id: 'app.components.Login.username.help',
      defaultMessage: 'Please enter your username',
    },
  },
  password: {
    label: {
      id: 'app.components.Login.password.label',
      defaultMessage: 'Password',
    },
    placeholder: {
      id: 'app.components.Login.password.placeholder',
      defaultMessage: 'Password',
    },
    help: {
      id: 'app.components.Login.password.help',
      defaultMessage: 'Please enter your password',
    },
  },
});
