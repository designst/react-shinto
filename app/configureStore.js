/* @flow */

import { createLogger } from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import { routerMiddleware } from 'react-router-redux';
import { compose, createStore, applyMiddleware } from 'redux';

// import type { Store } from '../types';
import createReducer from './createReducer';

const sagaMiddleware = createSagaMiddleware();

export default (history, initialState) => {
  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  // 2. routerMiddleware: Syncs the location/URL path to the state
  const middlewares = [
    sagaMiddleware,
    routerMiddleware(history),
  ];

  let composeEnhancers = compose;

  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    const loggerMiddleware = createLogger();
    middlewares.push(loggerMiddleware);

    // If Redux DevTools Extension is installed use it, otherwise use Redux compose
    /* eslint-disable no-underscore-dangle */
    // noinspection JSUnresolvedVariable
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
      // noinspection JSUnresolvedFunction
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        // Prevent recomputing reducers for `replaceReducer`
        shouldHotReload: false,
      });
    }
    /* eslint-enable */
  }

  const enhancers = [
    applyMiddleware(...middlewares),
  ];

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers)
  );

  // Extensions
  store.runSaga = sagaMiddleware.run;
  store.injectSagas = {}; // Saga registry
  store.injectReducers = {}; // Reducer registry

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./createReducer', () => {
      try {
        const nextReducer = require('./createReducer').default;

        store.replaceReducer(nextReducer);
      } catch (error) {
        console.error(`==> ðŸ˜­  Reducer hot reloading error ${error}`);
      }
    });
  }

  return store;
};
