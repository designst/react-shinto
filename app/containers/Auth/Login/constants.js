export const LOGIN_REQUEST = '@shinto/containers/auth/login/LOGIN_REQUEST';
export const LOGIN_SUCCESS = '@shinto/containers/auth/login/LOGIN_SUCCESS';
export const LOGIN_FAILURE = '@shinto/containers/auth/login/LOGIN_FAILURE';
