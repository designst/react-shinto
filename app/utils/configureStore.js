/* @flow */

import fp from 'lodash/fp';
import thunk from 'redux-thunk';
import { init } from '@rematch/core';
import { createLogger } from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import createLoadingPlugin from '@rematch/loading';
import { routerMiddleware } from 'react-router-redux';
import createWaitForActionMiddleware from 'redux-wait-for-action';

import createReducer, { rootReducers } from './createReducer';
import createApiService, { createApiServicePlugin } from './apiService';

const loadingPluginOptions = {};
const loadingPlugin = createLoadingPlugin(loadingPluginOptions);

export const createFakeReducers = initialState => {
  const fakeReducer = (state = {}) => state;
  const rootReducerKeys = Object.keys(rootReducers);

  return Object.keys(initialState)
    .filter(key => !rootReducerKeys.includes(key))
    .reduce((reducers, key) => {
      reducers[key] = fakeReducer;
      return reducers;
    }, {});
};

export default (history, initialState, configuration = {}) => {
  // Create the store with three middlewares
  // 1. thunkMiddleware: Makes thunk work
  // 2. sagaMiddleware: Makes redux-sagas work
  // 3. routerMiddleware: Syncs the location/URL path to the state

  const { auth: { token } = {} } = initialState;
  const apiService = createApiService(token);

  const sagaMiddleware = createSagaMiddleware({
    context: {
      apiService,
    },
  });

  const waitForActionMiddleware = createWaitForActionMiddleware();

  const middlewares = [thunk, sagaMiddleware, waitForActionMiddleware, routerMiddleware(history)];

  if (__DEV__ && __CLIENT__) {
    const loggerMiddleware = createLogger();
    middlewares.push(loggerMiddleware);
  }

  // Create fake reducers which are not already known by the injectedReducers map
  // but which are expected by the server-side rendered initialState object.
  // The created fakeReducers will be replaced after client-side reducers are injected.
  const initialStateReducers = createFakeReducers(initialState);

  const store = init(
    fp.assign(
      {
        redux: {
          middlewares,
          initialState,
          reducers: rootReducers,
          combineReducers: rematchReducers => {
            const { injectedReducers = {} } = store || {};
            // Override rematch combineReducers to handle the fakeReducers which are reducers
            // not already known but expected by the initialState object and to handle the
            // already injectedReducers.

            return createReducer(
              // injected modelReducers are already created by the rematch model function,
              // so we can ignore them here to avoid additional loops.
              {},
              {
                ...rematchReducers,
                ...injectedReducers,
              },
              initialStateReducers,
            );
          },
        },
        plugins: [loadingPlugin, createApiServicePlugin(apiService)],
      },
      configuration,
    ),
  );

  apiService.store = store;

  // Extensions
  store.runSaga = sagaMiddleware.run;
  store.injectedSagas = {}; // Saga registry
  store.injectedModels = {}; // Model registry
  store.injectedReducers = {}; // Reducer registry
  store.initialStateReducers = initialStateReducers;

  /* istanbul ignore next */
  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./createReducer', () => {
      try {
        const createNextReducer = require('./createReducer').default;
        const nextReducer = createNextReducer(
          store.injectedModels,
          store.injectedReducers,
          store.initialStateReducers,
        );

        store.replaceReducer(nextReducer);
      } catch (error) {
        console.error(`==> Reducer hot reloading error ${error}`);
      }
    });
  }

  return store;
};
