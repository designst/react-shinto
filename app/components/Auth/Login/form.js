import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import { injectIntl, FormattedMessage } from 'react-intl';

import Button from 'material-ui/Button';
import { TextField } from 'redux-form-material-ui';

import messages from './messages';

const LoginForm = ({ intl, handleSubmit }) => (
  <form onSubmit={handleSubmit}>
    <Field
      fullWidth
      name="email"
      type="email"
      label={intl.formatMessage(messages.email.label)}
      margin="normal"
      component={TextField}
      helperText={intl.formatMessage(messages.email.help)}
      placeholder={intl.formatMessage(messages.email.placeholder)}
    />

    <Field
      fullWidth
      name="username"
      label={intl.formatMessage(messages.username.label)}
      margin="normal"
      component={TextField}
      helperText={intl.formatMessage(messages.username.help)}
      placeholder={intl.formatMessage(messages.username.placeholder)}
    />

    <Field
      fullWidth
      name="password"
      type="password"
      label={intl.formatMessage(messages.password.label)}
      margin="normal"
      component={TextField}
      helperText={intl.formatMessage(messages.password.help)}
      placeholder={intl.formatMessage(messages.password.placeholder)}
    />

    <Button type="submit" color="primary" variant="raised">
      <FormattedMessage {...messages.action} />
    </Button>
  </form>
);

LoginForm.propTypes = {
  intl: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

const withIntl = injectIntl(LoginForm);

export default reduxForm({
  form: 'login',
})(withIntl);
