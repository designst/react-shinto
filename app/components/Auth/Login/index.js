/**
 * Login
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import LoginForm from './form';
import messages from './messages';

const Login = ({ onSubmit }) => (
  <div className="b-login">
    <FormattedMessage {...messages.header} />

    <LoginForm onSubmit={onSubmit} />
  </div>
);

Login.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

export default Login;
