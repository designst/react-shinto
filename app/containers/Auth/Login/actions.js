import { LOGIN_REQUEST, LOGIN_SUCCESS, LOGIN_FAILURE } from './constants';

export const loginRequest = ({ username, password }) => ({
  type: LOGIN_REQUEST,
  username,
  password,
});

export const loginSuccess = data => ({
  type: LOGIN_SUCCESS,
  data,
});

export const loginFailure = error => ({
  type: LOGIN_FAILURE,
  error,
});
