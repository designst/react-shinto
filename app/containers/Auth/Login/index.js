import { compose } from 'redux';
import { connect } from 'react-redux';

import Login from 'components/Auth/Login';

import { loginRequest } from './actions';

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  onSubmit: loginRequest,
};

const connector = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(connector)(Login);
