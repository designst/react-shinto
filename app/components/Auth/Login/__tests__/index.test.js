import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { StaticRouter, memoryHistory } from 'react-router-dom';

import configureStore from 'utils/configureStore';

import Login from '../index';

describe('<Login />', () => {
  test('renders', () => {
    const store = configureStore(memoryHistory, {});
    const tree = renderer
      .create(
        <Provider store={store}>
          <IntlProvider locale="en">
            <StaticRouter context={{}}>
              <Login onSubmit={jest.fn()} />
            </StaticRouter>
          </IntlProvider>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
