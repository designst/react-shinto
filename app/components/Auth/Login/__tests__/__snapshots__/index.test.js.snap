// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders 1`] = `
<div
  className="b-login"
>
  <span>
    This is the Login component!
  </span>
  <form
    onSubmit={[Function]}
  >
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
      onBlur={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-10 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9"
        data-shrink={false}
        htmlFor={undefined}
      >
        Username
      </label>
      <div
        className="MuiInput-root-15 MuiInput-fullWidth-22 MuiInput-formControl-16 MuiInput-underline-19"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-23"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="username"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder="Username"
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="text"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root-28"
        id={undefined}
      >
        Please enter your username
      </p>
    </div>
    <div
      aria-describedby={undefined}
      className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
      onBlur={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
    >
      <label
        className="MuiFormLabel-root-10 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9"
        data-shrink={false}
        htmlFor={undefined}
      >
        Password
      </label>
      <div
        className="MuiInput-root-15 MuiInput-fullWidth-22 MuiInput-formControl-16 MuiInput-underline-19"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-23 MuiInput-inputType-26"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder="Password"
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="password"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root-28"
        id={undefined}
      >
        Please enter your password
      </p>
    </div>
    <button
      className="MuiButtonBase-root-47 MuiButton-root-32 MuiButton-raised-37 MuiButton-raisedPrimary-38"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="submit"
    >
      <span
        className="MuiButton-label-33"
      >
        <span>
          Login
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-49"
      />
    </button>
  </form>
</div>
`;
